### PR TITLE
Update log4j2. Clarify logging included artifacts in shading.

### PR DESCRIPTION
**Pull request by afs**

### CODE REVIEW

**Review by kinow — APPROVED** [review]

**Comment by afs** [comment]
Thanks - still seems good to wait a day or so before merging to be sure the outside world has quietened down. 😸

File: jena-fuseki2/jena-fuseki-geosparql/pom.xml [hunk 1]
```diff
@@ -98,6 +98,17 @@
...
 
+        <!-- Put in to ensure the dependency isn't optional. -->
+        <dependency>
+          <groupId>org.apache.logging.log4j</groupId>
+          <artifactId>log4j-api</artifactId>
+        </dependency>
+
+        <dependency>
+          <groupId>org.apache.logging.log4j</groupId>
+          <artifactId>log4j-core</artifactId>
```
Comment (berezovskyi) [inline jena-fuseki2/jena-fuseki-geosparql/pom.xml:109]: I think `<scope>runtime</scope>` can be added to this dependency and other `-impl` dependencies as logging code should only directly depend on the `api`.

Comment (afs) [reply jena-fuseki2/jena-fuseki-geosparql/pom.xml:109]: These are for the most-part packaging classes.

jena-fuseki-geosparql is both packing and code for the command line and Fuseki configuration.  Might be better to leave it as scope-compile but jena-fuseki-server should be scope-runtime as it's only shade-packaging.

Comment (afs) [reply jena-fuseki2/jena-fuseki-geosparql/pom.xml:109]: (pushed and change for scope-setting for jena-fuseki-server)

**Comment by afs** [comment]
Merged because of [CVE-2021-45046](https://nvd.nist.gov/vuln/detail/CVE-2021-45046) which is rated "moderate".

**Pull request merged**